lib/feature/one_hot_encoder.cc

```diff
@@ -109,35 +112,36 @@ std::string getOnlyFileInDirectory(std::string path) {
 }
 } // anonymous namespace
 
-llvm::Expected<OneHotEncoderModel>
-OneHotEncoderModel::load(const std::string path) {
-  OneHotEncoderModel model;
+tfrt::AsyncValueRef<OneHotEncoderModel>
+OneHotEncoderModel::load(const std::string path, tfrt::HostContext *host) {
+  tfrt::AsyncValueRef<OneHotEncoderModel> model =
+      MakeAvailableAsyncValueRef<OneHotEncoderModel>(&host);
 
   ifstream params_input(path + "/metadata");
   json params;
   params << params_input;
   std::string is_droplast = params["paramMap"]["dropLast"].get<std::string>();
-  model.setDropLast(is_droplast.compare("false"));
+  model->setDropLast(is_droplast.compare("false"));
   params_input.close();
 
   std::string model_data_filename = getOnlyFileInDirectory(path + "/data");
   if (!model_data_filename.compare("")) {
-    return tfrt::MakeStringError(
+    return tfrt::MakeErrorAsyncValueRef(
         "Failed to load OneHotEncoderModel: model data directory " + path +
         "/data does not exist, or it has zero or more than one file.");
   }
 
   ifstream model_data_input(path + "/data/" + model_data_filename);
   std::string model_data_str((std::istreambuf_iterator<char>(model_data_input)),
                              std::istreambuf_iterator<char>());
-  llvm::Error err = model.setModelData(model_data_str);
+  tfrt::AsyncValueRef<tfrt::Chain> err = model->setModelData(model_data_str);
   model_data_input.close();
 
-  if (err) {
-    return tfrt::MakeStringError(
+  if (err.IsError()) {
+    return tfrt::MakeErrorAsyncValueRef(
         "Failed to load OneHotEncoderModel: invalid model data file " + path +
         "/data/" + model_data_filename);
   }
 
-  return model;
+  return model.CopyRef();
```

Comment:
`CopyRef()` is useful when the original object will be used in 2+ ways. In this case, the `model` will no longer be used in this method after the `return ...`, so it will be more efficient (and intuitive) to just do `return model`.

Note that `return model` is effectively the same as `return std::move(model)` since `return ...` is the last statement in the method. This would trigger the move constructor of `model` (i.e. `tfrt::AsyncValueRef`).

Reply:
OK. I'll make the change.